### ISSUE TITLE
Batch insert cannot evaluate SQL expression

### ISSUE THREAD

**Issue by dqn** [jmoiron/sqlx#622]
Single record insert can use SQL expressions in the VALUES clause, but batch insert added in #285 cannot.

e.g.

```go
user := &User{Name: "foo"}
db.NamedExec("INSERT INTO users (name, created_at, updated_at) VALUES (:name, NOW(), NOW())", user) // ok

users := []*User{
	{Name: "bar"},
	{Name: "baz"},
}
db.NamedExec("INSERT INTO users (name, created_at, updated_at) VALUES (:name, NOW(), NOW())", users) // pq: syntax error at end of input
```

**Comment by rcornacchia** [comment]
I'm also running into an issue with bulk insert but inserting one row works.

**Comment by dmngu9** [comment]
can you post the generated query from it?

**Comment by dqn** [comment]
@dmngu9 This.

```go
user := &User{Name: "foo"}
s1, _, _ := db.BindNamed("INSERT INTO users (name, created_at, updated_at) VALUES (:name, NOW(), NOW())", user)
fmt.Println(s1) // => INSERT INTO users (name, created_at, updated_at) VALUES ($1, NOW(), NOW())

users := []*User{
	{Name: "bar"},
	{Name: "baz"},
}
s2, _, _ := db.BindNamed("INSERT INTO users (name, created_at, updated_at) VALUES (:name, NOW(), NOW())", users)
fmt.Println(s2) // => INSERT INTO users (name, created_at, updated_at) VALUES ($1, NOW(),($2, NOW(), NOW())
```

**Comment by dmngu9** [comment]
same issue as mine. Theres sth wrong with query with built in function . https://github.com/jmoiron/sqlx/issues/640

**Comment by dmngu9** [comment]
if you replace NOW() with pq.FormatTime(time.Now()), it would work. Something wrong with NOw() affect generate query string

**Comment by carmel** [comment]
I encountered a similar problem:
```golang
test := Test{
  Name: " John",
  Age:  30,
  Id:   "bskdvfjreo018g2c5pqg",
}
// table test has fields: id, name, age
_, err := sqlxdb.DB.NamedExec("NSERT INTO test(name,age,id)VALUES(:Name,:Age,:Id)", &test)
log.Println(err)

// log as follow:

// 2020/08/07 11:49:15 could not find name Name in &sqlx.Test{Id:"bskdvfjreo018g2c5pqg", Name:"Vector", CreateAt:0, Age:30}
```

**Comment by dqn** [comment]
The named query
`INSERT INTO users (name, created_at, updated_at) VALUES (:name, NOW(), NOW())`
will be compiled into
`INSERT INTO users (name, created_at, updated_at) VALUES (?, NOW(), NOW())`.

The VALUES clause is extracted by a regular expression as `(?, NOW()`.

It should be extracted by parsing, not by regular expressions. However, due to the nature of writing raw SQL and the need to deal with various dialects, this problem is not easy to resolve.

**Comment by jmoiron** [comment]
fixed in #667
